modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/mv/io/RowVersionDataIo.java

```diff
@@ -60,25 +60,54 @@ protected void writeRowData(long pageAddr, int dataOff, int payloadSize, RowVers
         putShort(addr, 0, (short) payloadSize);
         addr += 2;
 
-        addr += Timestamps.writeTimestamp(addr, 0, row.timestamp());
+        addr += Timestamps.writeTimestampToMemory(addr, 0, row.timestamp());
 
         addr += PartitionlessLinks.writeToMemory(addr, row.nextLink());
 
-        ByteBuffer value = row.value();
-        putInt(addr, 0, value.limit());
+        putInt(addr, 0, row.valueSize());
         addr += 4;
 
-        putByteBuffer(addr, 0, value);
+        putByteBuffer(addr, 0, row.value());
     }
 
     /** {@inheritDoc} */
     @Override
     protected void writeFragmentData(RowVersion row, ByteBuffer buf, int rowOff, int payloadSize) {
```

Comment:
Should we validate the `rowOff` against the `row` capacity and `payloadSize` against the `buf` capacity?

Reply:
I think these assertions should be present in the page memory core (that calls this method), not in this method, because they would check the page memory core correctness and not the correctness of this specific code